src/wfs/src/test/java/org/geoserver/wfs/TransactionTest.java

```diff
@@ -438,4 +438,88 @@ public void testUpdateWithDifferentPrefix() throws Exception {
+                + "</cgf:Lines>" + "</wfs:Insert>"
+                + "<wfs:Insert > " + "<cgf:Lines>" + "<cgf:lineStringProperty>"
+                + "<gml:LineString>" + "<gml:coordinates decimal=\".\" cs=\",\" ts=\" \">"
+                + "494475.71056417,5433016.8189320 494982.70115668,5435041.95096615"
+                + "</gml:coordinates>" + "</gml:LineString>"
+                + "</cgf:lineStringProperty>" + "<cgf:id>t0002</cgf:id>"
+                + "</cgf:Lines>" + "</wfs:Insert>"
+                + "<wfs:Insert > " + "<cgf:Lines>" + "<cgf:lineStringProperty>"
+                + "<gml:LineString>" + "<gml:coordinates decimal=\".\" cs=\",\" ts=\" \">"
+                + "494475.71056414,5433016.8189327 494982.70115660,5435041.95096615"
+                + "</gml:coordinates>" + "</gml:LineString>"
+                + "</cgf:lineStringProperty>" + "<cgf:id>t0002</cgf:id>"
+                + "</cgf:Lines>" + "</wfs:Insert>"
+                + "<wfs:Insert > " + "<cgf:Lines>" + "<cgf:lineStringProperty>"
+                + "<gml:LineString>" + "<gml:coordinates decimal=\".\" cs=\",\" ts=\" \">"
+                + "494475.71056419,5433016.8189327 494982.70115666,5435041.95096611"
+                + "</gml:coordinates>" + "</gml:LineString>"
+                + "</cgf:lineStringProperty>" + "<cgf:id>t0002</cgf:id>"
+                + "</cgf:Lines>" + "</wfs:Insert>"
+                + "<wfs:Insert > " + "<cgf:Lines>" + "<cgf:lineStringProperty>"
+                + "<gml:LineString>" + "<gml:coordinates decimal=\".\" cs=\",\" ts=\" \">"
+                + "494475.71056417,5433016.8189329 494982.70115667,5435041.95096610"
+                + "</gml:coordinates>" + "</gml:LineString>"
+                + "</cgf:lineStringProperty>" + "<cgf:id>t0002</cgf:id>"
+                + "</cgf:Lines>" + "</wfs:Insert>"
+                + "<wfs:Insert > " + "<cgf:Lines>" + "<cgf:lineStringProperty>"
+                + "<gml:LineString>" + "<gml:coordinates decimal=\".\" cs=\",\" ts=\" \">"
+                + "494475.71056419,5433016.8189329 494982.70115665,5435041.95096620"
+                + "</gml:coordinates>" + "</gml:LineString>"
+                + "</cgf:lineStringProperty>" + "<cgf:id>t0002</cgf:id>"
+                + "</cgf:Lines>" + "</wfs:Insert>"
+                + "<wfs:Insert > " + "<cgf:Lines>" + "<cgf:lineStringProperty>"
+                + "<gml:LineString>" + "<gml:coordinates decimal=\".\" cs=\",\" ts=\" \">"
+                + "494475.710564120,5433016.8189323 494982.70115655,5435041.95096670"
+                + "</gml:coordinates>" + "</gml:LineString>"
+                + "</cgf:lineStringProperty>" + "<cgf:id>t0002</cgf:id>"
+                + "</cgf:Lines>" + "</wfs:Insert>"
+                + "</wfs:Transaction>";
+        Document dom = postAsDOM("wfs", features);
+        print(dom);
+        XMLAssert.assertXpathEvaluatesTo("new11", "//wfs:InsertResult/ogc:FeatureId[last()]/@fid", dom);
```

Comment:
I guess to do that we would need a GetFeature using the last fid, and double check the attribute values.

If the datastore used for this test would respect the provided FIDs this would be much easier. Do we know what is behind this test?